lib/logstash/plugin_mixins/http_client.rb

```diff
@@ -75,6 +75,9 @@ def self.included(base)
       # See https://hc.apache.org/httpcomponents-client-ga/httpclient/apidocs/org/apache/http/impl/conn/PoolingHttpClientConnectionManager.html#setValidateAfterInactivity(int)[these docs for more info]
       base.config :validate_after_inactivity, :validate => :number, :default => 200
 
+      # Enable/disable the SSL configurations
+      base.config :ssl_enabled, :validate => :boolean, :default => true
```

Comment:
[es-output](https://www.elastic.co/guide/en/logstash/current/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch-ssl_enabled) and [es-input](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-elasticsearch.html#plugins-inputs-elasticsearch-ssl_enabled) have `ssl_enabled` with no default value. It bases on the host prefix, https or http, to enable ssl feature, which is kind of setting to `auto`.

Would default to true changes the behaviour of existing plugins? 
Could the mixin default to `auto` and preserve the current smart ssl feature?

Reply:
sorry, I mixed the http plugin with es plugin 🤦 The settings make perfect sense